homeassistant/components/wemo/__init__.py

```diff
@@ -144,6 +144,10 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
 
     dispatcher = wemo_data.config_entry_data.dispatcher
     if unload_ok := await dispatcher.async_unload_platforms(hass):
+        for coordinator in list(
+            wemo_data.config_entry_data.device_coordinators.values()
+        ):
+            await coordinator.async_shutdown()
```

Comment:
Wemo was relying on side effect of the first forwarded platform config entry forwards calling the `async_on_onload` callbacks which had the side effect of the shutdown on the coordinator before the config entry was unloaded.